vspec/__init__.py

```diff
@@ -247,13 +252,34 @@ def cleanup_deep_model(deep_model):
     if "$name$" in deep_model:
         del deep_model['$name$']
 
-    if (deep_model["type"] == "branch"):
+    # children as of today exists only for branches
+    if "children" in deep_model:
```

Comment:
This change is not needed now, but needed later when we start supporting structs as we then want to clean items as well I think